### PR TITLE
Test/IDN-62 login viewmodel test

### PR DESCRIPTION
**Pull request by ShroukMohamed16**
PR Description:

This PR introduces unit tests for the `Login ViewModel` to ensure its logic behaves correctly under different scenarios.

Changes included:

- Added test cases for all public functions of the ViewModel.
- Verified state updates and StateFlow emissions.
- Covered success, error, and edge cases to ensure robust behavior.

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -65,8 +66,20 @@ koin-annotations = { module = "io.insert-koin:koin-annotations", version.ref = "
 koin-ksp-compiler = { module = "io.insert-koin:koin-ksp-compiler", version.ref = "koin-annotations" }
 
 #Unit testing
+kotlin-test = { module = "org.jetbrains.kotlin:kotlin-test", version.ref = "kotlin" }
```
Comment (nabilBouzineDev) [inline gradle/libs.versions.toml:69]: Please @ShroukMohamed16, avoid using duplicate alias names like `kotlin-test` it's already added above and same thing for others as well

**Review by nabilBouzineDev — CHANGES_REQUESTED** [review]
